GithubRepo/Data/Repository/GithubMainRepository.swift

```diff
@@ -6,47 +6,37 @@
 //
 
 import Foundation
+import RxRelay
+import RxSwift
 
 class GithubMainRepository: GithubRepository {
     private let dataSource: GithubDataSource
-    private(set) var repositories: [Repository]
-    private(set) var loading: Bool
-    private(set) var error: Bool
-    weak var delegate: GithubRepositoryDelegate?
+    private(set) var repositories: BehaviorRelay<[Repository]>
+    private(set) var state: BehaviorRelay<FetchState>
 
     init(dataSource: GithubDataSource = RemoteGithubDataSource()) {
         self.dataSource = dataSource
-        self.repositories = []
-        self.loading = false
-        self.error = false
+        self.repositories = BehaviorRelay<[Repository]>(value: [])
+        self.state = BehaviorRelay<FetchState>(value: .inital)
     }
 
     func handleChangeRepositories(_ repo: [Repository]) {
-        self.repositories = repo
-        delegate?.didChangeRepositories(repositories: repo)
+        self.repositories.accept(repo)
```

Comment:
once you refactor datasource/service in rx code, you will not need this kind of functions like `handleSomething` and updating the subject.
You can directly connect the results of some observables in your subject/relay.. this is the magic of Rx